dracut/modules.d/99asahi-firmware/install-asahi-firmware.sh

```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
+# SPDX-License-Identifier: MIT
+
+type getarg > /dev/null 2>&1 || . /lib/dracut-lib.sh
+
+info ":: Asahi: Installing vendor firmware to root filesystem..."
+if [ ! -d /sysroot/lib/firmware/vendor ]; then
+    warn ":: Asahi: Vendor firmware directory missing on the root filesystem!"
```

Comment:
If we got this far, we know we're on an Apple Silicon machine. If this directory does not exist at this point, it should be created so that the firmware can be copied for consumption by drivers regardless of the status of userspace packages (e.g. for bootable installer images).

Reply:
I imagine the concern here might be that the rootfs could be mounted read-only, though I don't know whether that is the case here. Not creating the dir here is valid logic, but then there has to be a package that owns its creation.

Reply:
Yup, at this stage the root filesystem is still read only, at least for Fedora.

Reply:
Yup, that's the case for dracut, and it's not safe to do remounts, so I don't do that.